Docstring additions and typo fixes

Nudies opened the pull request:
improve code readability in _parse_decls

untitaker commented:
Some typo fixes are legit, but some things are not necessary. `and-or` is just a style choice.

untitaker commented on click/core.py:
```diff
@@ -1425,7 +1425,7 @@ class Option(Parameter):
                          help page.  Normally, defaults are not shown.
     :param prompt: if set to `True` or a non empty string then the user will
-                   be prompted for input if not set.  If set to `True` the
+                   be prompted for input.  If set to `True` the
```
This sentence still doesn't make sense.

untitaker commented:
@Nudies Any update?

Nudies commented:
@untitaker Apologies seems I missed your original comment 27days ago. I reverted back the `and-or` change, sorry for making that assumption. Also cleaned up the `Option` class docstring a little.

Nudies commented on click/core.py:
```diff
@@ -1424,9 +1424,9 @@ class Option(Parameter):
-                   prompt will be the option name capitalized.
+    :param prompt: if set to `True` or a non empty string then the user will be
+                   prompted for input.  If set to `True` the prompt will be the
+                   option name capitalized.
```
I cleaned this up a little. Is the sentence still confusing? If so what part?

untitaker commented:
Cool, thanks!
